### PR TITLE
This is the translation file in Arabic

### PR DESCRIPTION
**Pull request by foodmaher**
**This is the translation file in Arabic**

This is a withdrawal request to add Arabic to this project.
 Warning: I have not tried try the translation file before, if it is possible to examine the file by someone here, I will be grateful to him to know how to show the Arabic texts within the application and also examines the variables if they work or no because the Arabic reflect the English texts also and make them uncoordinated and this is a problem The variables are not called.

**Modified:** Because of incomprehensible speech and bad English.

### CODE REVIEW

**Comment by corbinlc** [comment]
Thanks for your submission.  We will look this over.  It may be a few days.

**Comment by foodmaher** [comment]
@corbinlc 
Thank you for your quick response
I would like to check the speech format check and the variable name %1$s and %2$s does it work or not
How do I update my fork?

**Comment by corbinlc** [comment]
Team, we should look at this on Monday and provide some feedback.  

@foodmaher are you trying to pull our latest changes into your fork?  If so, this should help with that: https://help.github.com/en/articles/syncing-a-fork

Anyone reading this... our team does not know any Arabic, so we will probably just double check things with Google Translate or similar.  If anyone else is reading this and is fluent in English and Arabic, please feel free to provide a review.

**Comment by foodmaher** [comment]
@corbinlc 
# Question
the question already had to take the latest changes for me fork. Thanks for giving the answer
# As for the translation

The Arabic translation is understandable and does not need to be reviewed but the problem is in checking the variables names and if they are written correctly. Unfortunately, I do not know anyone here who speaks Arabic to help us, but I do not think there is a problem with translation. It can be a problem in defining variables.

So I asked to check the file only to make sure of the variables because, as I said Arabic make the names of variables are irregular because they are written from the opposite side, so you need to check to see if they were defined correctly and.

**Can you link the translation file with the trial version so that I can check the translation file through my phone or send me an apk file with the translation file**

File: app/src/main/res/values-ar/strings.xml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    <string name="services_notification_channel_description">الخدمات المستمرة</string>
+    <string name="service_notification_title">UserLAnd</string>
+    <string name="service_notification_description">يدير UserLAnd خدمة خلفية.</string>
+
+    <!-- Progress bar messages -->
+    <string name="progress_start_step">الحصول على الأمور جاهزة بالنسبة لك &#8230;</string>
+    <string name="progress_fetching_asset_lists">جلب قوائم الأصول &#8230;</string>
+    <string name="progress_checking_for_required_updates">التحقق مما إذا كانت هناك حاجة إلى التحديثات &#8230;</string>
+    <string name="progress_downloading">تنزيل الأصول المطلوبة &#8230;</string>
+    <string name="progress_downloading_out_of">تم الانتهاء من ٪1$d من ٪2$d</string>
```
Comment (luongthomas) [inline app/src/main/res/values-ar/strings.xml]: I've tested this string in Android Studio and on my device and I noticed it isn't showing correctly.  I found the cause.

The cause is that the `٪` symbol that this string has, is not the same one that Kotlin is expecting when substituting a variable into the string.  Please replace `٪` with `%` and it should behave correctly.

File: app/src/main/res/values-ar/strings.xml [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    <string name="progress_copying_downloads">نسخ الملفات التي تم تنزيلها إلى التخزين المحلي &#8230;</string>
+    <string name="progress_setting_up_filesystem">إعداد نظام الملفات &#8230;</string>
+    <string name="progress_extraction_details">الاستخراج: ٪1$s</string>
```
Comment (luongthomas) [inline app/src/main/res/values-ar/strings.xml]: Refer to the comment on line 97 about replacing `٪` with `%`

File: app/src/main/res/values-ar/strings.xml [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    <string name="error_app_description_not_found">لا يمكن العثور على وصف لهذا التطبيق. حاول تحديث.</string>
+    <string name="error_vnc_password_length_incorrect">يجب أن تكون كلمة مرور VNC بين 6 إلى 8 أحرف!</string>
+    <string name="error_username_invalid">اسم المستخدم يحتوي على أحرف غير صالحة</string>
+    <string name="error_password_invalid">كلمة السر بها أحرف غير صالحة</string>
+    <string name="error_vnc_password_invalid">كلمة السر VNC تحتوي  أحرف غير صالحة</string>
+    <string name="error_username_invalid_characters">يجب أن تتكون الأحرف في اسم المستخدم من 1 إلى 30 حرفًا بأحرف و / أو أرقام صغيرة.</string>
+    <string name="error_username_in_blacklist">ربما يتم حجز اسم المستخدم هذا بواسطة Linux ولا يمكن استخدامه كاسم مستخدم لنظام الملفات</string>
+
+    <!-- Illegal State Dialog -->
+    <string name="illegal_state_message">٪1$s \n سيتم الآن تعطل التطبيق بحيث يمكننا استلام سجل أخطاء. لا تتردد أيضًا في إخبارنا على جيثب! نحن نعمل على حل هذه القضايا.</string>
```
Comment (luongthomas) [inline app/src/main/res/values-ar/strings.xml]: Refer to the comment on line 97 about replacing `٪` with `%`

File: app/src/main/res/values-ar/strings.xml [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    <string name="illegal_state_title">لقد دخل UserLAnd حالة غير قانونية!</string>
+
+    <string name="illegal_state_transition">انتقال الحالة السيئة:٪1$s</string>
```
Comment (luongthomas) [inline app/src/main/res/values-ar/strings.xml]: Refer to the comment on line 97 about replacing `٪` with `%`

File: app/src/main/res/values-ar/strings.xml [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+    <string name="illegal_state_no_filesystem_selected_when_credentials_selected">تم إرسال بيانات الاعتماد نظام الملفات، ولكن لم يتم تحديدها.</string>
+    <string name="illegal_state_no_app_selected_when_preference_submitted">تم إرسال تفضيل التطبيق ، ولكن لم يتم اختيار واحد.</string>
+    <string name="illegal_state_no_app_selected_when_preparation_started">جارٍ محاولة بدء إعداد التطبيق ، ولكن لم يتم تحديد أي تطبيق.</string>
+    <string name="illegal_state_error_fetching_app_database_entries">لا يمكن جلب إدخالات قاعدة بيانات التطبيقات.</string>
+    <string name="illegal_state_error_copying_app_script">لا يمكن نسخ البرنامج النصي للتطبيق.</string>
+
+    <string name="illegal_state_no_session_selected_when_preparation_started">محاولة التعامل مع إعداد الجلسة قبل اختيار واحد.</string>
+    <string name="illegal_state_error_fetching_asset_lists">فشل في استرداد قوائم الأصول.</string>
+    <string name="illegal_state_downloads_did_not_complete_successfully">فشلت التنزيلات بسبب:٪1$s.</string>
```
Comment (luongthomas) [inline app/src/main/res/values-ar/strings.xml]: Refer to the comment on line 97 about replacing `٪` with `%`

File: app/src/main/res/values-ar/strings.xml (hunk 1 above)
Comment (luongthomas) [reply app/src/main/res/values-ar/strings.xml]: As it is, it shows up like this when downloading the filesystem assets.

screenshot_20190226-110801_userland

**Comment by foodmaher** [comment]
@luongthomas I noticed that you referred to some of the texts as (Outdated) Do you want me delete it? or what? and do there are new texts not found in the file to add?

Thank you ... 😀😀

File: app/src/main/res/values-ar/strings.xml (hunk 2 above)
Comment (foodmaher) [reply app/src/main/res/values-ar/strings.xml]: ok

File: app/src/main/res/values-ar/strings.xml (hunk 1 above)
Comment (foodmaher) [reply app/src/main/res/values-ar/strings.xml]: fixed porblem

**Comment by luongthomas** [comment]
Thanks @foodmaher!

**Comment by foodmaher** [comment]
@luongthomas Thank you ... 😘😘😘

**Pull request merged**